pkg/controller/nodes/subworkflow/launchplan/launchplan.go

```diff
@@ -31,6 +31,9 @@ type Executor interface {
 	// Kill a remote execution
 	Kill(ctx context.Context, executionID *core.WorkflowExecutionIdentifier, reason string) error
 
+	// Get the definition of a launch plan. This is primarily used to ensure all the TypedInterfaces match up before actually executing.
+	GetLaunchPlan(ctx context.Context, launchPlanRef *core.Identifier) (*admin.LaunchPlan, error)
```

Comment:
I think we should create a separate interface for this. And then pass that interface to Dynamic task. This is weird coupling - subworkflow only ever launches a launchplan, having this method is confusing and in the future may lead to weird uses. But having a separate interface will prevent these problems, and also make uses cases clear